ampd/src/broadcast/broadcaster.rs

```diff
@@ -202,11 +195,27 @@ where
     ///
     /// * `Error::EstimateGas` - If the gas estimation fails
     pub async fn estimate_gas(&mut self, msgs: Vec<Any>) -> Result<Gas> {
-        let acc_sequence = self.acc_sequence.read().await;
+        let mut acc_sequence = self.acc_sequence.write().await;
 
-        cosmos::estimate_gas(&mut self.client, msgs, self.pub_key, *acc_sequence)
-            .await
-            .change_context(Error::EstimateGas)
+        let res =
+            match cosmos::estimate_gas(&mut self.client, msgs.clone(), self.pub_key, *acc_sequence)
+                .await
+            {
+                Ok(gas) => Ok(gas),
+                Err(e) => match parse_sequence_error(e.to_string()) {
+                    // Retry with the expected sequence number
+                    Some(expected_seq) => {
+                        *acc_sequence = expected_seq;
+
+                        cosmos::estimate_gas(&mut self.client, msgs, self.pub_key, *acc_sequence)
+                            .await
+                    }
```

Comment:
what is the actual root cause of the issue? is there a possibility that the expected is also not correct, because something changes before the next broadcast? would be helpful to describe the problem in the comment here

Reply:
added a description of possible reasons why a mismatch could happen. The root cause was that the previous fix was resetting the sequence each time a message was added to the queue, but there were still unconfirmed transactions, so the account sequence expected for simulation didn't match the account sequence that was fetch when querying the account

Reply:
how is the sequence determined in usual operations?

Reply:
when the broadcaster is instantiated, the account is queried to get the sequence number:
https://github.com/axelarnetwork/axelar-amplifier/blob/07027683949a25a59956b54787721eb788e0b9c8/ampd/src/broadcast/broadcaster.rs#L138-L148
Then after each successful broadcast, the sequence number is incremented:
https://github.com/axelarnetwork/axelar-amplifier/blob/07027683949a25a59956b54787721eb788e0b9c8/ampd/src/broadcast/broadcaster.rs#L259-L264